Wilab-io/sample-apps
# [Document feeding] Separate from deploy and use certificates

# Description

## Token removal

An input in the Connection settings will allow the user to upload its api-key file and avoid using tokens.

## Deploy and feed logic separation

Now when the deployment is performed, no document processing is done.

## Document management

To upload documents, the application must be deployed, from there, you can delete and upload documents without re-deploying the application.
The application also has to be deployed to delete documents. Now, the deletion logic removes the document from Vespa Cloud.
There are modals for document uploading and deletion, and their error alternatives. The process is now slower than before so the user needs to know what's going on.

# How to test

1. Run the DB separately.
2. Run the app and log in.
3. Assert that you cannot upload documents if the application isn't deployed.
4. Assert that you cannot deploy the application if an API key is not uploaded in the Connections tab.
5. Assert that the deployment process is now faster.
6. Assert that the document uploading now feeds the Vespa Cloud Environment.
7. Assert that the document deletion also deletes the document from Vespa (it shouldn't appear in the search results).

Closes #63
Closes #64
Closes #65
Closes #56
Closes #57
